src/lib.rs

```diff
@@ -416,81 +672,120 @@ impl Timer {
 /// schedule is cancelled.
 #[derive(Clone)]
 pub struct Guard {
-    should_execute: Arc<AtomicBool>
+    should_execute: Arc<AtomicBool>,
+    ignore_drop: bool
 }
 impl Guard {
     fn new() -> Self {
         Guard {
-            should_execute: Arc::new(AtomicBool::new(true))
+            should_execute: Arc::new(AtomicBool::new(true)),
+            ignore_drop: false
         }
     }
     fn should_execute(&self) -> bool {
         self.should_execute.load(AtomicOrdering::Relaxed)
     }
+    pub fn ignore(mut self) {
+        self.ignore_drop = true;
+    }
 }
 impl Drop for Guard {
     /// Cancel a schedule.
     fn drop(&mut self) {
-        self.should_execute.store(false, AtomicOrdering::Relaxed)
+        if !self.ignore_drop {
+            self.should_execute.store(false, AtomicOrdering::Relaxed)
+        }
     }
 }
 
-#[test]
-fn test_schedule_with_delay() {
-    let timer = Timer::new();
-    let (tx, rx) = channel();
-    let mut guards = vec![];
-
-    // Schedule a number of callbacks in an arbitrary order, make sure
-    // that they are executed in the right order.
-    let mut delays = vec![1, 5, 3, -1];
-    let start = UTC::now();
-    for i in delays.clone() {
-        println!("Scheduling for execution in {} seconds", i);
-        let tx = tx.clone();
-        guards.push(timer.schedule_with_delay(Duration::seconds(i), move || {
-            println!("Callback {}", i);
-            tx.send(i).unwrap();
-        }));
-    }
-
-    delays.sort();
-    for (i, msg) in (0..delays.len()).zip(rx.iter()) {
-        let elapsed = (UTC::now() - start).num_seconds();
-        println!("Received message {} after {} seconds", msg, elapsed);
-        assert_eq!(msg, delays[i]);
-        assert!(delays[i] <= elapsed && elapsed <= delays[i] + 3, "We have waited {} seconds, expecting [{}, {}]", elapsed, delays[i], delays[i] + 3);
-    }
-
-    // Now make sure that callbacks that are designed to be executed
-    // immediately are executed quickly.
-    let start = UTC::now();
-    for i in vec![10, 0] {
-        println!("Scheduling for execution in {} seconds", i);
-        let tx = tx.clone();
-        guards.push(timer.schedule_with_delay(Duration::seconds(i), move || {
-            println!("Callback {}", i);
-            tx.send(i).unwrap();
-        }));
-    }
-
-    assert_eq!(rx.recv().unwrap(), 0);
-    assert!(UTC::now() - start <= Duration::seconds(1));
-}
+#[cfg(test)]
+mod tests {
```

Comment:
While the underlying implementation is identical, it would be reassuring to have tests also for `MessageTimer` (besides the non-cloning).